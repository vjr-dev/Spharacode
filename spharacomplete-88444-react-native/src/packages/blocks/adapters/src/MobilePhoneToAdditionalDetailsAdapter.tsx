import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class MobilePhoneToAdditionalDetailsAdapter {
  send: (message: Message) => void;

  constructor() {
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.NavigationMobilePhoneAdditionalDetailsMessage)
    ]);
  }

  convert = (from: Message): Message => {
    const to = new Message(getName(MessageEnum.NavigationMessage));

    to.addData(
      getName(MessageEnum.NavigationTargetMessage),
      'AdditionalDetailForm'
    );

    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );

    const raiseMessage: Message = new Message(
      getName(MessageEnum.NavigationPayLoadMessage)
    );
    raiseMessage.addData(
      getName(MessageEnum.AuthTokenDataMessage),
      from.getData(getName(MessageEnum.AuthTokenDataMessage))
    );

    to.addData(getName(MessageEnum.NavigationRaiseMessage), raiseMessage);

    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
