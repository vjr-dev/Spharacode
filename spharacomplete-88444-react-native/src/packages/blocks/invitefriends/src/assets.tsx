export const invite_mail = require("../assets/mail.png");
export const invite_sms = require("../assets/sms.png");
export const invite_whatsapp = require("../assets/whatsapp.png");
export const invite_friend = require("../assets/invite_friend.png");
