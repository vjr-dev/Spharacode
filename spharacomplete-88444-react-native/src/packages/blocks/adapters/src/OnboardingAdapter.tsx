import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class OnboardingAdapter {
  send: (message: Message) => void;

  constructor() {
    // const uuidv4 = require('uuid/v4');
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.OnboardingGuideControllerDoneOnboarding),
      getName(MessageEnum.OnboardingGuideControllerSkipOnboarding)
    ]);
  }

  convert = (from: Message): Message => {
    runEngine.debugLog("OnboardingAdapter", from.id)
    const to = new Message(getName(MessageEnum.NavigationMessage));
    to.addData(getName(MessageEnum.NavigationTargetMessage), 'Home');
    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );
    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
