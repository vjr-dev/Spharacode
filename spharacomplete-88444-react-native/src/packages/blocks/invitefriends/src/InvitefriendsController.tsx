import { IBlock } from "../../../framework/src/IBlock";
import { Message } from "../../../framework/src/Message";
import { BlockComponent } from "../../../framework/src/BlockComponent";
import { runEngine } from "../../../framework/src/RunEngine";

// Customizable Area Start
import {
  Linking,
  Platform
} from "react-native";
// Customizable Area End

export const configJSON = require("./config");

export interface Props {
  navigation: any;
  id: string;
  // Customizable Area Start
  // Customizable Area End
}

interface S {
  // Customizable Area Start
  // Customizable Area End
}

interface SS {
  id: any;
}

export default class InvitefriendsController extends BlockComponent<
  Props,
  S,
  SS
> {
  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);

    // Customizable Area Start
    this.subScribedMessages = [];

    this.state = {
    };
    // Customizable Area End
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }

  async receive(from: string, message: Message) {
    // Customizable Area Start
    runEngine.debugLog("Message Recived", message);
    // Customizable Area End
  }

  // Customizable Area Start
  handleMailAction = () => {
    let phoneOS = Platform.OS === "android" ? "?" : "&";
    let mailTo = `mailto:foo@example.com?cc=${phoneOS}subject=yourSubject${phoneOS}body=yourMessage`;
    if (Linking.canOpenURL(mailTo)) {
      Linking.openURL(mailTo);
    }
  };

  handleSMSAction = () => {
    let phoneOS = Platform.OS === "android" ? "?" : "&";
    let sms = `sms:8885555512${phoneOS}body=This is a Dummy Message`;
    if (Linking.canOpenURL(sms)) {
      Linking.openURL(sms);
    }
  };

  handleWhatsAppAction = () => {
    let whatsapp = `whatsapp://send?phone='8885555512'&text='Hello Message'`;
    if (Linking.canOpenURL(whatsapp)) {
      Linking.openURL(whatsapp);
    }
  };

  btnEmailIconProps = {
    onPress: () => {
      this.handleMailAction();
    }
  };

  btnWhatsAppIconProps = {
    onPress: () => {
      this.handleWhatsAppAction();
    }
  };

  btnSMSIconProps = {
    onPress: () => {
      this.handleSMSAction();
    }
  };
  // Customizable Area End

}
