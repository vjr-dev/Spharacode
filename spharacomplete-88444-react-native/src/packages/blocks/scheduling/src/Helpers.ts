/* App/Lib/GeneralHelpers.js */
import { Platform, StyleSheet } from 'react-native';
import { heightFromPercentage } from '../../../framework/src/Utilities';

// Colors Constants
export const COLORS = {
  TRANSPARENT: 'transparent',
  white: 'rgb(255,255,255)',
  black: 'rgb(0,0,0)',
  black10: 'rgb(10,10,10)',
  green: 'rgb(76,217,100)',
  red: 'red',
  yellow: 'rgba(255, 212, 40, 1)',
  darkGray: 'rgba(105,105,105,0.9)',
  lightGray: '#a7b6b9',
  skipGray: 'rgb(91,91,92)',
  borderColor: 'rgb(240, 240, 240)',
  inactiveIndicator: 'rgb(228, 233, 242)',
  inputLabel: 'rgb(143,155,179)',
  inputIcon: 'rgb(197,206,224)',
  inputValue: 'rgb(46,58,89)',
  discountRed: 'rgb(199,5,28)',
  orange: 'rgb(255,157,43)',
  infoGray: 'rgb(163,164,165)',
  ReviewBlack: 'rgb(34,43,69)',
  blue: 'blue',
  blueGrey: 'rgb(143,155,179)',
  darkBlueGrey: 'rgb(46,58,89)',
  lightBlueGrey: 'rgb(197, 206, 224)'
};

export const CommonStyle = StyleSheet.create({
  container: {
    flex: 1,
    width: "100%",
    justifyContent: "flex-start",
    alignItems: "center"
  },
  wrapper: {
    flex: 1,
    width: "100%",
    height: "100%"
  },
  headerContainer:{
    backgroundColor:COLORS.yellow,
    height: '10%',
  },
  authHeader:{
    height: heightFromPercentage(10),
    backgroundColor: COLORS.yellow
  },
  authHeaderWhite:{
    height: heightFromPercentage(10),
    backgroundColor: COLORS.white
  },
  formContainer: {
    width: "80%"
  },
  textAlignCenter:{
    textAlign: 'center'
  },
  labelTextStyle: {
    color: COLORS.inputLabel
  },
  inputContainerStyle:{
    borderBottomWidth: 1,
    borderBottomColor: COLORS.inactiveIndicator
  },
  error: {
    color: COLORS.red,
    fontSize: 14
  },
  scrollStyle: {
    width: "100%"
  },
  btnTitleStyle:{
    fontSize: 17,
    color: COLORS.black
  },
  closeButtonStyle: {
    backgroundColor: COLORS.inputIcon,
    height: 40,
    width: 100,
    borderRadius: 8
  },
  yellowButtonStyle: {
    backgroundColor: COLORS.yellow,
    height: 40,
    width: 100,
    borderRadius: 8
  },
  closeBtnTitleStyle:{
    color: COLORS.black
  },
  scrollContainer: {
    flexGrow: 1,
    alignItems: "center"
  },
  shadowStyle: {
    shadowOffset:
      Platform.OS === "ios"
        ? { width: 0, height: 1 }
        : { width: 0, height: 2 },
    shadowColor: COLORS.inputLabel,
    shadowOpacity: 0.9,
    shadowRadius: Platform.OS === "ios" ? 3 : 5,
    elevation: 5
  },
  boxShadowStyle: {
    backgroundColor: COLORS.white,
    marginHorizontal:2,
    marginVertical:1,
    shadowOffset:
    Platform.OS === "ios"
      ? { width: 0, height: 1 }
      : { width: 0, height: 2 },
    shadowColor: COLORS.inputLabel,
    shadowOpacity: 0.9,
    shadowRadius: Platform.OS === "ios" ? 1 : 5,
    elevation: 2,    
    borderRadius:8,
  },
  toastModal: {
    borderRadius: 20,
    marginHorizontal: 15,
    paddingHorizontal: '12%',
    paddingVertical: 100,
    alignContent: "center",
    justifyContent: "center",
  },
  clearModal: {
    borderRadius: 5,
    marginHorizontal: 50,
    height: 220,
    backgroundColor: COLORS.red
  },
  imageThumbContain: {
    resizeMode: "contain",
    width: "100%",
    height: "100%"
  },
  imageThumbCover: {
    resizeMode: "cover",
    width: "100%",
    height: "100%"
  },
  flexContainer: {
    flex: 1,
    width: "100%"
  },
  rowStyle: {
    flexDirection: "row"
  },
  flexStart: {
    justifyContent: "flex-start"
  },
  flexEnd: {
    justifyContent: "flex-end"
  },
  jCCenter: {
    justifyContent: "center"
  },
  spaceBetween: {
    justifyContent: "space-between",    
  },
  spaceAround: {
    justifyContent: 'space-around'
  },
  alignItemsCenter:{
    alignItems: "center"
  },
  styleCenter: {
    justifyContent: "flex-end",
    alignItems: "center"
  },
  centerStyle: {
    justifyContent: "center",
    alignItems: "center"
  },
  width100P:{
    width: '100%'
  },
  subView: {
    flex: 1,
    width: "100%",
    height: "100%"
  },
  emptyList: {
    alignItems: "center",
    marginTop: 100
  },
  emptyListFilter: {
    alignItems: "center",
    justifyContent: "center",
    marginTop: 100
  },
  backIconView: {
    marginVertical: 10,
    marginHorizontal: 5,
    width: 30,
    height: 30,
    justifyContent: "flex-end",
    alignItems: "flex-end"
  },
  crossImage: {
    tintColor: COLORS.red,
    width: "100%",
    height: "100%",
    resizeMode: "contain"
  },
  searchTextInput: {
    width: "80%"
  },
  errorReport: {
    width: "70%",
    color: COLORS.red,
    fontSize: 15,
    marginLeft: "28.8%"
  }
});