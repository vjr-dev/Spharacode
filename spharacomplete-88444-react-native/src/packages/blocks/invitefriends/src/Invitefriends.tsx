import React from "react";
// Customizable Area Start
import {
  StyleSheet,
  View,
  Text,
  Image,
  TouchableOpacity
} from "react-native";
import * as IMG_CONST from "./assets";
import scale, { verticalScale } from "../../../components/src/Scale";
// Customizable Area End

import InvitefriendsController from "./InvitefriendsController";

export default class InviteFriends extends InvitefriendsController {
  render() {
    // Customizable Area Start
    return (
      <View style={styles.container}>
        <View>
          <Text testID={"textInstructions"} style={styles.TextStyleMsg}>
            Share your code with friends and get an exciting offers
          </Text>
          <Text testID={"textTitle"} style={styles.TextStyle}>
            Invite your friends
          </Text>
        </View>
        <View style={styles.row}>
          <TouchableOpacity
            testID={"btnEmailIcon"}
            {...this.btnEmailIconProps}
          >
            <Image
              testID={"imgEmailIcon"}
              source={IMG_CONST.invite_mail}
              style={styles.inviteImage}
            />
          </TouchableOpacity>
          <TouchableOpacity
            testID={"btnWhatsAppIcon"}
            {...this.btnWhatsAppIconProps}
          >
            <Image
              testID={"imgWhatsAppIcon"}
              source={IMG_CONST.invite_whatsapp}
              style={styles.inviteWPImage}
            />
          </TouchableOpacity>
          <TouchableOpacity
            testID={"btnSMSIcon"}
            {...this.btnSMSIconProps}
          >
            <Image
              testID={"imgSMSIcon"}
              source={IMG_CONST.invite_sms}
              style={styles.inviteImage}
            />
          </TouchableOpacity>
        </View>

        <View style={styles.bottomView}>
          <Image
            testID={"imgPosterIcon"}
            source={IMG_CONST.invite_friend}
            style={styles.bottomImageView}
          />
        </View>
      </View>
    );
  }
  // Customizable Area End
}

// Customizable Area Start
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "white"
  },
  TextStyleMsg: {
    color: "black",
    marginTop: verticalScale(35),
    fontSize: scale(19.7),
    textAlign: "center"
  },
  TextStyle: {
    color: "#366ef9",
    marginTop: verticalScale(52.3),
    textAlign: "center",
    fontSize: scale(23.3)
  },
  bottomView: {
    width: "100%",
    height: verticalScale(209.3),
    marginBottom: verticalScale(20.7),
    position: "absolute",
    bottom: 0
  },
  bottomImageView: {
    width: "100%",
    height: verticalScale(209.3),
    resizeMode: "contain"
  },
  row: {
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    marginTop: verticalScale(52.3)
  },
  inviteImage: {
    width: scale(53.7),
    height: scale(53.7)
  },
  inviteWPImage: {
    width: scale(53.7),
    height: scale(53.7),
    marginStart: scale(25),
    marginEnd: scale(25)
  }
});
// Customizable Area End
