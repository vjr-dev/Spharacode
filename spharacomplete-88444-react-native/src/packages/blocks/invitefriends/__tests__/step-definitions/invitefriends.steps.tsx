import { defineFeature, loadFeature } from "jest-cucumber";
import { shallow, ShallowWrapper } from "enzyme";

import * as helpers from "../../../../framework/src/Helpers";
import React from "react";
import Invitefriends from "../../src/Invitefriends";
const navigation = require("react-navigation");

const screenProps = {
  navigation: navigation,
  id: "Invitefriends"
};

const feature = loadFeature(
  "./__tests__/features/invitefriends-scenario.feature"
);

defineFeature(feature, test => {
  beforeEach(() => {
    jest.resetModules();
    jest.doMock("react-native", () => ({ Platform: { OS: "web" } }));
    jest.spyOn(helpers, "getOS").mockImplementation(() => "web");
  });

  test("User navigates to invitefriends", ({ given, when, then }) => {
    let exampleBlockA: ShallowWrapper;
    let instance: Invitefriends;

    given("I am a User loading invitefriends", () => {
      exampleBlockA = shallow(<Invitefriends {...screenProps} />);

      expect(exampleBlockA).toBeTruthy();
      expect(exampleBlockA).toMatchSnapshot();

      instance = exampleBlockA.instance() as Invitefriends;
    });

    when("I navigate to the invitefriends", () => {
      instance = exampleBlockA.instance() as Invitefriends;
    });

    then("invitefriends will load with out errors", () => {
      expect(exampleBlockA).toBeTruthy();
      expect(exampleBlockA).toMatchSnapshot();
    });

    then("I can enter text with out errors", () => {
      expect(exampleBlockA).toBeTruthy();
      expect(exampleBlockA).toMatchSnapshot();
    });

    then("I can select the button with with out errors", () => {
      let buttonComponent = exampleBlockA.findWhere(
        node => node.prop("testID") === "btnEmailIcon"
      );
      buttonComponent.simulate("press");
      buttonComponent = exampleBlockA.findWhere(
        node => node.prop("testID") === "btnWhatsAppIcon"
      );
      buttonComponent.simulate("press");
      buttonComponent = exampleBlockA.findWhere(
        node => node.prop("testID") === "btnSMSIcon"
      );
      buttonComponent.simulate("press");
      expect(exampleBlockA).toMatchSnapshot();
    });

    then("I can leave the screen with out errors", () => {
      instance.componentWillUnmount();
      expect(exampleBlockA).toBeTruthy();
      expect(exampleBlockA).toMatchSnapshot();
    });
  });
});
