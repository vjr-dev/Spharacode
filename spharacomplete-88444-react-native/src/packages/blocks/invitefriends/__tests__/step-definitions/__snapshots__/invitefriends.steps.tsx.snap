// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invitefriends User navigates to invitefriends 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 35.688405797101446,
          "marginTop": 63.4375,
          "textAlign": "center",
        }
      }
      testID="textInstructions"
    >
      Share your code with friends and get an exciting offers
    </Text>
    <Text
      style={
        Object {
          "color": "#366ef9",
          "fontSize": 42.210144927536234,
          "marginTop": 94.79374999999999,
          "textAlign": "center",
        }
      }
      testID="textTitle"
    >
      Invite your friends
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 94.79374999999999,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="btnEmailIcon"
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/blocks/invitefriends/assets/mail.png",
          }
        }
        style={
          Object {
            "height": 97.28260869565219,
            "width": 97.28260869565219,
          }
        }
        testID="imgEmailIcon"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="btnWhatsAppIcon"
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/blocks/invitefriends/assets/whatsapp.png",
          }
        }
        style={
          Object {
            "height": 97.28260869565219,
            "marginEnd": 45.289855072463766,
            "marginStart": 45.289855072463766,
            "width": 97.28260869565219,
          }
        }
        testID="imgWhatsAppIcon"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="btnSMSIcon"
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/blocks/invitefriends/assets/sms.png",
          }
        }
        style={
          Object {
            "height": 97.28260869565219,
            "width": 97.28260869565219,
          }
        }
        testID="imgSMSIcon"
      />
    </TouchableOpacity>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "height": 379.35625000000005,
        "marginBottom": 37.51875,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../packages/blocks/invitefriends/assets/invite_friend.png",
        }
      }
      style={
        Object {
          "height": 379.35625000000005,
          "resizeMode": "contain",
          "width": "100%",
        }
      }
      testID="imgPosterIcon"
    />
  </View>
</View>
`;

exports[`invitefriends User navigates to invitefriends 2`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 35.688405797101446,
          "marginTop": 63.4375,
          "textAlign": "center",
        }
      }
      testID="textInstructions"
    >
      Share your code with friends and get an exciting offers
    </Text>
    <Text
      style={
        Object {
          "color": "#366ef9",
          "fontSize": 42.210144927536234,
          "marginTop": 94.79374999999999,
          "textAlign": "center",
        }
      }
      testID="textTitle"
    >
      Invite your friends
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 94.79374999999999,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="btnEmailIcon"
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/blocks/invitefriends/assets/mail.png",
          }
        }
        style={
          Object {
            "height": 97.28260869565219,
            "width": 97.28260869565219,
          }
        }
        testID="imgEmailIcon"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="btnWhatsAppIcon"
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/blocks/invitefriends/assets/whatsapp.png",
          }
        }
        style={
          Object {
            "height": 97.28260869565219,
            "marginEnd": 45.289855072463766,
            "marginStart": 45.289855072463766,
            "width": 97.28260869565219,
          }
        }
        testID="imgWhatsAppIcon"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="btnSMSIcon"
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/blocks/invitefriends/assets/sms.png",
          }
        }
        style={
          Object {
            "height": 97.28260869565219,
            "width": 97.28260869565219,
          }
        }
        testID="imgSMSIcon"
      />
    </TouchableOpacity>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "height": 379.35625000000005,
        "marginBottom": 37.51875,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../packages/blocks/invitefriends/assets/invite_friend.png",
        }
      }
      style={
        Object {
          "height": 379.35625000000005,
          "resizeMode": "contain",
          "width": "100%",
        }
      }
      testID="imgPosterIcon"
    />
  </View>
</View>
`;

exports[`invitefriends User navigates to invitefriends 3`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 35.688405797101446,
          "marginTop": 63.4375,
          "textAlign": "center",
        }
      }
      testID="textInstructions"
    >
      Share your code with friends and get an exciting offers
    </Text>
    <Text
      style={
        Object {
          "color": "#366ef9",
          "fontSize": 42.210144927536234,
          "marginTop": 94.79374999999999,
          "textAlign": "center",
        }
      }
      testID="textTitle"
    >
      Invite your friends
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 94.79374999999999,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="btnEmailIcon"
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/blocks/invitefriends/assets/mail.png",
          }
        }
        style={
          Object {
            "height": 97.28260869565219,
            "width": 97.28260869565219,
          }
        }
        testID="imgEmailIcon"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="btnWhatsAppIcon"
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/blocks/invitefriends/assets/whatsapp.png",
          }
        }
        style={
          Object {
            "height": 97.28260869565219,
            "marginEnd": 45.289855072463766,
            "marginStart": 45.289855072463766,
            "width": 97.28260869565219,
          }
        }
        testID="imgWhatsAppIcon"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="btnSMSIcon"
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/blocks/invitefriends/assets/sms.png",
          }
        }
        style={
          Object {
            "height": 97.28260869565219,
            "width": 97.28260869565219,
          }
        }
        testID="imgSMSIcon"
      />
    </TouchableOpacity>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "height": 379.35625000000005,
        "marginBottom": 37.51875,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../packages/blocks/invitefriends/assets/invite_friend.png",
        }
      }
      style={
        Object {
          "height": 379.35625000000005,
          "resizeMode": "contain",
          "width": "100%",
        }
      }
      testID="imgPosterIcon"
    />
  </View>
</View>
`;

exports[`invitefriends User navigates to invitefriends 4`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 35.688405797101446,
          "marginTop": 63.4375,
          "textAlign": "center",
        }
      }
      testID="textInstructions"
    >
      Share your code with friends and get an exciting offers
    </Text>
    <Text
      style={
        Object {
          "color": "#366ef9",
          "fontSize": 42.210144927536234,
          "marginTop": 94.79374999999999,
          "textAlign": "center",
        }
      }
      testID="textTitle"
    >
      Invite your friends
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 94.79374999999999,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="btnEmailIcon"
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/blocks/invitefriends/assets/mail.png",
          }
        }
        style={
          Object {
            "height": 97.28260869565219,
            "width": 97.28260869565219,
          }
        }
        testID="imgEmailIcon"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="btnWhatsAppIcon"
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/blocks/invitefriends/assets/whatsapp.png",
          }
        }
        style={
          Object {
            "height": 97.28260869565219,
            "marginEnd": 45.289855072463766,
            "marginStart": 45.289855072463766,
            "width": 97.28260869565219,
          }
        }
        testID="imgWhatsAppIcon"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="btnSMSIcon"
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/blocks/invitefriends/assets/sms.png",
          }
        }
        style={
          Object {
            "height": 97.28260869565219,
            "width": 97.28260869565219,
          }
        }
        testID="imgSMSIcon"
      />
    </TouchableOpacity>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "height": 379.35625000000005,
        "marginBottom": 37.51875,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../packages/blocks/invitefriends/assets/invite_friend.png",
        }
      }
      style={
        Object {
          "height": 379.35625000000005,
          "resizeMode": "contain",
          "width": "100%",
        }
      }
      testID="imgPosterIcon"
    />
  </View>
</View>
`;

exports[`invitefriends User navigates to invitefriends 5`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 35.688405797101446,
          "marginTop": 63.4375,
          "textAlign": "center",
        }
      }
      testID="textInstructions"
    >
      Share your code with friends and get an exciting offers
    </Text>
    <Text
      style={
        Object {
          "color": "#366ef9",
          "fontSize": 42.210144927536234,
          "marginTop": 94.79374999999999,
          "textAlign": "center",
        }
      }
      testID="textTitle"
    >
      Invite your friends
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 94.79374999999999,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="btnEmailIcon"
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/blocks/invitefriends/assets/mail.png",
          }
        }
        style={
          Object {
            "height": 97.28260869565219,
            "width": 97.28260869565219,
          }
        }
        testID="imgEmailIcon"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="btnWhatsAppIcon"
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/blocks/invitefriends/assets/whatsapp.png",
          }
        }
        style={
          Object {
            "height": 97.28260869565219,
            "marginEnd": 45.289855072463766,
            "marginStart": 45.289855072463766,
            "width": 97.28260869565219,
          }
        }
        testID="imgWhatsAppIcon"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="btnSMSIcon"
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/blocks/invitefriends/assets/sms.png",
          }
        }
        style={
          Object {
            "height": 97.28260869565219,
            "width": 97.28260869565219,
          }
        }
        testID="imgSMSIcon"
      />
    </TouchableOpacity>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "height": 379.35625000000005,
        "marginBottom": 37.51875,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../packages/blocks/invitefriends/assets/invite_friend.png",
        }
      }
      style={
        Object {
          "height": 379.35625000000005,
          "resizeMode": "contain",
          "width": "100%",
        }
      }
      testID="imgPosterIcon"
    />
  </View>
</View>
`;
